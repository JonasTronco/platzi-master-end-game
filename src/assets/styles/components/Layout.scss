@import '../main.scss';

.layout {
  @include colCenSta;

  &__container {
    @include flexCenSta;
    width: 1024px;
    padding: $space_x9 $space_x12;

    &__content {
      width: calc(100% - 214px);
    }
  }
}

@media screen and (max-width: 1024px) {
  .layout {
    &__container {
      @include colCenSta;
      width: 100%;
      padding: $space_x9 $space_x12;

      &__content {
        width: 100%;
      }
    }
  }
}

@media screen and (max-width: 425px) {
  .layout {
    &__container {
      @include colCenSta;
      flex-direction: column-reverse;
      width: 100%;
      padding: 0;
      padding-top: $space_x6;

      &__content {
        width: 90%;
        padding-bottom: $space_x6;
      }
    }
  }
}
