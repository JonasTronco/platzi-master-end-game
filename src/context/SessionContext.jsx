import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
