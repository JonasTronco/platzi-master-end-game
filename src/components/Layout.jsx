import PropTypes from 'prop-types';
import { useState, useEffect } from 'react';
import axios from 'axios';
import { useSession } from '../context/SessionContext';
import Header from './Header';
import Navbar from './Navbar';
import Modal from './Modal';
import MessageIcon from '../assets/icons/alarm.svg';
import '../assets/styles/components/Layout.scss';

function Layout({ children }) {
  const { session } = useSession();
  const [message, setMessage] = useState(false);
  const [messageResponse, setMessageResponse] = useState([]);

  function closeModal() {
    setMessage(false);
    setMessageResponse([]);
  }

  // eslint-disable-next-line consistent-return
  async function openModalMessage() {
    console.log('times');
    try {
      const URL = `${process.env.NEXT_PUBLIC_API_URL}/messages?patientId=${session.user.id}`;
      const config = {
        headers: { Authorization: `Bearer ${session.token}` },
      };
      const res = await axios.get(URL, config);
      if (res.data) {
        setMessageResponse(res.data.data);
        setMessage(true);
      }
    } catch (err) {
      return '';
    }
  }

  useEffect(() => {
    if (session && session.user.typeOfUser === 'Patient') {
      // eslint-disable-next-line consistent-return
      setInterval(() => {
        openModalMessage();
      }, 10000);
    }
  });

  return (
    <>
      <div className="layout">
        <Header />
        <div className="layout__container">
          <Navbar />
          <div className="layout__container__content">{children}</div>
        </div>
      </div>
      <Modal isOpen={message}>
        <div className="message">
          <MessageIcon className="message__icon" />
          { messageResponse.map((res) => (
            <strong className="message__text">{res.messageText}</strong>
          ))}
          <button className="btn" type="button" onClick={closeModal}>
            Accept
          </button>
        </div>
      </Modal>
    </>
  );
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
