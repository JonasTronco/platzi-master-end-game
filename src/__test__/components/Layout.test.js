import Layout from '../../components/Layout';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import { SessionProvider } from '../../context/SessionContext';
describe('Layout Component', () => {
  test('Should render Layout component', () => {
    const component = create(
      <SessionProvider>
        <Layout>
          <div></div>
        </Layout>
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('<Layout />', () => {
    const component = mount(
      <SessionProvider>
        <Layout>
          <div></div>
        </Layout>
      </SessionProvider>
    );
    expect(component.length).toEqual(1);
  });
});
