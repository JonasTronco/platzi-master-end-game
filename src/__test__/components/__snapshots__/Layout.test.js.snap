// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component Should render Layout component 1`] = `
<div
  className="layout"
>
  <header
    className="header"
  >
    <div
      className="header__logo"
    />
    <strong
      className="header__text"
    >
      HALAH
      <br />
      Laboratories
    </strong>
  </header>
  <div
    className="layout__container"
  >
    <div
      className="loader"
    />
    <div
      className="layout__container__content"
    />
  </div>
</div>
`;
